custom/enikshay/tests/test_adherence_updates.py

```diff
@@ -0,0 +1,291 @@
+import pytz
+from django.test import TestCase
+
+from corehq.apps.domain.models import Domain
+from corehq.apps.fixtures.models import FixtureDataType, FixtureTypeField, \
+    FixtureDataItem, FieldList, FixtureItemField
+from corehq.form_processor.tests.utils import FormProcessorTestUtils
+from custom.enikshay.tasks import *
+from .utils import *
+
+
+class TestAdherenceUpdater(TestCase):
+    @classmethod
+    def setUpClass(cls):
+        super(TestAdherenceUpdater, cls).setUpClass()
+        cls.domain = 'adherence-enikshay-test'
+        cls.setupFixtureData()
+        FormProcessorTestUtils.delete_all_cases()
+
+    def setUp(self):
+        super(TestAdherenceUpdater, self).setUp()
+        delete_all_users()
+        self.factory = CaseFactory(domain=self.domain)
+        self.user = CommCareUser.create(
+            self.domain,
+            "jon-snow@user",
+            "123",
+        )
+        self.person_id = u"person"
+        self.occurrence_id = u"occurrence"
+        self.episode_id = u"episode"
+        FormProcessorTestUtils.delete_all_cases()
+        self.case_updater = PeriodicCaseUpdater(self.domain)
+
+    @classmethod
+    def setupFixtureData(cls):
+        cls.fixture_data = {
+            'schedule1': '7',
+            'schedule2': '14',
+            'schedule3': '21',
+        }
+        cls.data_type = FixtureDataType(
+            domain=cls.domain,
+            tag=DAILY_SCHEDULE_FIXTURE_NAME,
+            name=DAILY_SCHEDULE_FIXTURE_NAME,
+            fields=[
+                FixtureTypeField(
+                    field_name=SCHEDULE_ID_FIXTURE,
+                    properties=[]
+                ),
+                FixtureTypeField(
+                    field_name="doses_per_week",
+                    properties=[]
+                ),
+            ],
+            item_attributes=[],
+        )
+        cls.data_type.save()
+        cls.data_items = []
+        for _id, value in cls.fixture_data.iteritems():
+            data_item = FixtureDataItem(
+                domain=cls.domain,
+                data_type_id=cls.data_type.get_id,
+                fields={
+                    SCHEDULE_ID_FIXTURE: FieldList(
+                        field_list=[
+                            FixtureItemField(
+                                field_value=_id,
+                            )
+                        ]
+                    ),
+                    "doses_per_week": FieldList(
+                        field_list=[
+                            FixtureItemField(
+                                field_value=value,
+                            )
+                        ]
+                    )
+                },
+                item_attributes={},
+            )
+            data_item.save()
+            cls.data_items.append(data_item)
+
+    def tearDown(self):
+        delete_all_users()
```

Comment:
nit: I think this could be `self.user.delete()` which might be slightly less work